tools/nl/embeddings/finetuning/finetune.py

```diff
@@ -297,64 +294,70 @@ def main(_):
+    # Checkpoint (save/upload) that model to GCS.
 
-    # Step 1. Loading the model.
+    # Step 1. Loading the base model.
     print(f"Loading the base model: {FLAGS.model_name_v2}")
     model_base = SentenceTransformer(FLAGS.model_name_v2)
 
     # Step 2a. Fine tuning with alternatives.
     print(f"(Intermediate) Fine tuning with alternatives. Stage: {FLAGS.stage}")
-    model_intermediate_finetuned = finetune_model(
+    model_intermediate = finetune_model(
         model_base, _generate_training_examples_from_alternatives(df_svs))
 
     ctx = utils.Context(gs=gs,
-                        model=model_intermediate_finetuned,
+                        model=model_intermediate,
                         bucket=bucket,
                         tmp='/tmp')
 
     # Step 2b. Upload the alternatives finetuned model to the NL model server's GCS bucket.
-    model_intermediate_folder_name = _save_finetuned_model(
-        ctx, "intermediate", FLAGS.model_name_v2)
+    model_intermediate_name = _save_finetuned_model(ctx, "intermediate",
+                                                    FLAGS.model_name_v2)
     print(
-        f"Produced and uploaded finetuned intermediate model: {model_intermediate_folder_name}"
+        f"Produced and uploaded finetuned intermediate model: {model_intermediate_name}"
     )
 
-  else:
+  elif FLAGS.pretuned_model:
+    # If a pretuned (intermediate) model was provided, use it to download and load
+    # the intermediate model.
+
     # Step 1. Loading the Intermediate pre-finetuned model.
     # No need for Steps 2a and 2b (see above).
-    model_intermediate_folder_name = FLAGS.pretuned_model
+    model_intermediate_name = FLAGS.pretuned_model
 
-    print(
-        f"Using the intermediate model (on GCS): {model_intermediate_folder_name}"
-    )
-    assert "." in model_intermediate_folder_name
-    assert STAGE_INTERMEDIATE in model_intermediate_folder_name
-    assert STAGE_FINAL not in model_intermediate_folder_name
+    print(f"Using the intermediate model (on GCS): {model_intermediate_name}")
+    assert "." in model_intermediate_name
+    assert STAGE_INTERMEDIATE in model_intermediate_name
+    assert STAGE_FINAL not in model_intermediate_name
 
     print(
-        f"Loading the pre-finetuned Intermediate model: {model_intermediate_folder_name}"
+        f"Loading the pre-finetuned Intermediate model: {model_intermediate_name}"
     )
     ctx = utils.Context(gs=gs, model=None, bucket=bucket, tmp='/tmp')
     downloaded_model_path = utils.download_model_from_gcs(
-        ctx, model_intermediate_folder_name)
-    model_intermediate_finetuned = SentenceTransformer(downloaded_model_path)
+        ctx, model_intermediate_name)
+    model_intermediate = SentenceTransformer(downloaded_model_path)
+
+  else:
+    # The intermediate model is the base model.
+    model_intermediate = SentenceTransformer(FLAGS.model_name_v2)
+    model_intermediate_name = FLAGS.model_name_v2
```

Comment:
When the intermediate model is the base model, do we need to call `_generate_training_examples_from_alternatives` too ?

Reply:
Ahh, I see... maybe that's also worth documenting.

Reply:
Added a couple of sentences to the comments.